app/models/orders_package.rb

```diff
@@ -20,4 +22,95 @@ def set_initial_state
       transition :requested => :designated
     end
   end
+
+  def self.update_partially_designated_item(package)
+    orders_package = OrdersPackage.find(package[:orders_package_id].to_i)
+    total_quantity = orders_package.quantity + package[:quantity].to_i
+    if(orders_package.state == "cancelled")
+      orders_package.update(quantity: total_quantity, state: "designated")
+    else
+      orders_package.update(quantity: total_quantity)
+    end
+  end
+
+  def self.undesignate_partially_designated_item(packages)
+    packages.each do |package|
+      quantity_to_reduce = package.last[:quantity].to_i
+      orders_package = OrdersPackage.find(package.last[:orders_package_id].to_i)
+      total_quantity = orders_package.quantity - quantity_to_reduce
+      if total_quantity == 0
+        orders_package.update(quantity: total_quantity, state: "cancelled")
+      else
+        orders_package.update(quantity: total_quantity, state: "designated")
+      end
+    end
+  end
+
+  def self.add_partially_designated_item(package)
+    self.create(
+      order_id: package[:order_id].to_i,
+      package_id: package[:package_id].to_i,
+      quantity: package[:quantity].to_i,
+      updated_by: User.current_user,
+      state: "designated"
+      )
+  end
+
+  def self.find_packages(order_id, package_id)
+    orders_packages = OrdersPackage.where("order_id = ? and package_id = ?", order_id, package_id)
+  end
+
+  def self.find_records(order_id)
```

Comment:
Agree re renaming but I also think this method could be refactored into a `scope` via has_many relationship instead?

    scope :by_order_id(order_id), -> { where(order_id: order_id) }